File: src/#590/solution.js
function Node(val, children) {
  this.val = val
  this.children = children
}

var postorder = function (root) {
  var arr = []
  helper(root)
  return arr

  function helper(node) {
    if (node === null) {
      return
    }

    for (let i = 0; i < node.children.length; i++) {
      helper(node.children[i])
    }
    arr.push(node.val)
  }
}

File: src/#590/solution.go
package solution

type Node struct {
	Val      int
	Children []*Node
}

func postorder(root *Node) []int {
	var arr []int

	helper(root,&arr)

	return arr
}

func helper(node *Node, arr *[]int) {
	if node == nil {
		return
	}
	for _, n := range node.Children {
		helper(n, arr)
	}
	*arr = append(*arr, node.Val)
}
